packages/dds/tree/src/simple-tree/api/identifierIndex.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { UsageError } from "@fluidframework/telemetry-utils/internal";
+import type { TreeIndexKey } from "../../feature-libraries/index.js";
+import type { ImplicitFieldSchema } from "../schemaTypes.js";
+import type { TreeNode } from "../core/index.js";
+import { ObjectNodeSchema } from "../objectNodeTypes.js";
+import type { TreeView } from "./tree.js";
+import { walkFieldSchema } from "../walkFieldSchema.js";
+import { createSimpleTreeIndex, type SimpleTreeIndex } from "./simpleTreeIndex.js";
+
+/**
+ * An index that returns tree nodes given their associated identifiers.
+ *
+ * @alpha
+ */
+export type IdentifierIndex = SimpleTreeIndex<string, TreeNode>;
+
+function isStringKey(key: TreeIndexKey): key is string {
+	return typeof key === "string";
+}
+
+/**
+ * Creates an {@link IdentifierIndex} for a given {@link TreeView}.
+ *
+ * @alpha
+ */
+export function createIdentifierIndex<TSchema extends ImplicitFieldSchema>(
+	view: TreeView<TSchema>,
+): IdentifierIndex {
+	// For each node schema, find which field key the identifier field is under.
+	// This can be done easily because identifiers are their own field kinds.
+	const identifierFields = new Map<string, string>();
+	walkFieldSchema(view.schema, {
+		node: (schemus) => {
+			if (schemus instanceof ObjectNodeSchema) {
+				for (const fieldKey of schemus.fields.keys()) {
+					identifierFields.set(schemus.identifier, fieldKey);
```

Comment:
I'm a bit confused. how do we know that this field is an identifier field here?

Reply:
we don't :)
this used to work correctly and at some point the check got removed, fixed it and added a test for it